src/map/pc.c

```diff
@@ -12311,6 +12311,50 @@ static void pc_check_supernovice_call(struct map_session_data *sd, const char *m
 	}
 }
 
+/**
+ * Sends a message t all online GMs having the specified permission.
+ *
+ * @param sender_name  Sender character name.
+ * @param permission   The required permission to receive this message.
+ * @param message      The message body.
+ *
+ * @return The amount of characters the message was delivered to.
+ */
+// The transmission of GM only Wisp/Page from server to inter-server
+static int pc_wis_message_to_gm(const char *sender_name, int permission, const char *message)
+{
+	nullpo_ret(sender_name);
+	nullpo_ret(message);
+	int mes_len = (int)strlen(message) + 1; // + null
+	int count = 0;
+
+	// information is sent to all online GM
+	map->foreachpc(pc->wis_message_to_gm_sub, permission, sender_name, message, mes_len, &count);
+
+	return count;
+}
+
+/**
+ * Helper function for pc_wis_message_to_gm().
+ */
+static int pc_wis_message_to_gm_sub(struct map_session_data *sd, va_list va)
+{
+	nullpo_ret(sd);
+
+	int permission = va_arg(va, int);
+	if (!pc_has_permission(sd, permission))
+		return 0;
+
+	const char *sender_name = va_arg(va, const char *);
```

Comment:
probably need check all pointers from va_arg to NULL?

Reply:
Added, thank you